src/gui/screen_menu_odometer.cpp

```diff
@@ -0,0 +1,61 @@
+// screen_menu .cpp
+
+#include <stdlib.h>
+
+#include "screen_menus.hpp"
+#include "screen_menu.hpp"
+#include "i18n.h"
+#include "odometer.hpp"
+
+static const constexpr HelperConfig HelpCfg = { 10, IDR_FNT_NORMAL };
+enum : int {
+    TEXT_MAX_LENGTH = 150
+};
+
+using Screen = ScreenMenu<EHeader::On, EFooter::On, HelpCfg, MI_RETURN>;
+
+class ScreenMenuOdometer : public Screen {
+    char text[TEXT_MAX_LENGTH];
```

Comment:
Why do you need to store the text in RAM at all? Can't it be computed when rendering the screen?